R/misc_cite_packages.R

```diff
@@ -17,20 +18,20 @@ show_packages <- function(session) {
   versions <- c()
   names <- c()
   for (pkg_name in names(pkgs)) {
-    pkg <- pkgs[[pkg_name]]
 
     citation <- format(citation(pkg_name))[[2]] %>%
       strsplit("\n") %>%
-      purrr::flatten() %>%
+      unlist() %>%
       paste(collapse = "SPLIT") %>%
-      strsplit("SPLITSPLIT")
+      strsplit("SPLITSPLIT") %>%
+      unlist()
 
     i <- 1
-    while (stringr::str_detect(citation[[1]][i], "To cite ")) {
+    while (stringr::str_detect(citation[i], "To cite ")) {
       i <- i + 1
     }
 
-    citation <- gsub("\\s", "", gsub("SPLIT", "", citation[[1]][i]))
+    citation <- gsub("  ", " ", trimws(gsub("\\s", " ", gsub("SPLIT", "", citation[i])), which="both"))
```

Comment:
Is `trimws()` needed here? I though `gsub("\\s")` will remove any whitespaces anyway?

```r
gsub("\\s", "", " this is a test!   ")
#> [1] "thisisatest!"
```

Reply:
Hum I guess you're right, I have to investigate more

Reply:
![image](https://user-images.githubusercontent.com/8875533/52984693-81f3fe00-342b-11e9-8625-6c9d2814a950.png)

It seems (the second line) that it does not work as intended, hence the trimws 🤔

Reply:
You're replacing white space with white space! :-D

Reply:
haha I don't want to remove ALL the whitespaces (want to keep those within the sentence 😄 )

Reply:
Oh I see... I don't have the former code in mind, but that was probably indeed too much removing the trimws()